From 816eefd28ca0c57a8088c352ea5472dfc2046abe Mon Sep 17 00:00:00 2001
From: Eldar Abusalimov <eldar.abusalimov@jetbrains.com>
Date: Wed, 18 Oct 2017 07:17:25 +0300
Subject: [PATCH] getpath: Don't use python.zip when running from build dir

This fixes a build error when rebuilding while having a
python.zip installed in $PREFIX.

Relates:
    https://bugs.python.org/issue21166 Bus error in pybuilddir.txt 'python -m sysconfigure --generate-posix-vars' build step
---
 Modules/getpath.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/Modules/getpath.c b/Modules/getpath.c
index 0f91643..84489bd 100644
--- a/Modules/getpath.c
+++ b/Modules/getpath.c
@@ -732,7 +732,8 @@ calculate_path(void)
         defpath = delim + 1;
     }
 
-    bufsz += wcslen(zip_path) + 1;
+    if (pfound >= 0 || efound >= 0)
+        bufsz += wcslen(zip_path) + 1;
     bufsz += wcslen(exec_prefix) + 1;
 
     buf = PyMem_New(wchar_t, bufsz);
@@ -749,9 +750,11 @@ calculate_path(void)
     else
         buf[0] = '\0';
 
-    /* Next is the default zip path */
-    wcscat(buf, zip_path);
-    wcscat(buf, delimiter);
+    if (pfound >= 0 || efound >= 0) {
+        /* Next is the default zip path */
+        wcscat(buf, zip_path);
+        wcscat(buf, delimiter);
+    }
 
     /* Next goes merge of compile-time $PYTHONPATH with
      * dynamically located prefix.
-- 
2.7.4

